test/e2e/scheduling/priorities.go

```diff
@@ -219,16 +219,27 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
 		//we need apply more taints on a node, because one match toleration only count 1
 		ginkgo.By("Trying to apply 10 taint on the nodes except first one.")
 		nodeName := nodeList.Items[0].Name
-
+		var totalTaints []*v1.Taint
 		for index, node := range nodeList.Items {
 			if index == 0 {
 				continue
 			}
 			for i := 0; i < 10; i++ {
-				testTaint := addRandomTaitToNode(cs, node.Name)
-				defer framework.RemoveTaintOffNode(cs, node.Name, *testTaint)
+				testTaint := addRandomTaintToNode(cs, node.Name)
+				totalTaints = append(totalTaints, testTaint)
 			}
 		}
+		defer func() {
+			for index, node := range nodeList.Items {
```

Comment:
A more robust approach that would tolerate addition/removal of nodes is to iterate over `totalTaints` and remove them. In order to do so, you will need to change the slice to a map:

`addedTaints map[string][]*v1.Taint`

Key of the map will be a node name.

Reply:
Ok, I will address them, thanks @bsalamat